### ISSUE TITLE
False negative when class derives from two generic base classes that conflict by type argument

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5748]
Here's a minimal repro of this problem.

```python
class X(Mapping[str, int], Sequence[int]):
    ...
```

This should generate an error because `Mapping[str, int]` is a subtype of `Sequence[str]` which is incompatible with `Sequence[int]`.

This is related to [this issue in pylance](https://github.com/microsoft/pylance-release/issues/1295#issue-892717386) and [this issue in mypy](https://github.com/python/mypy/issues/10485).

**Comment by erictraut** [comment]
This is included in pyright 1.1.324, which I just published. It will also be included in a future release of pylance.

**Comment by Trimatix** [comment]
Many thanks for this Eric!

Directed at the general community, I have a question regarding usage:
Is there a way to indicate that type parameters which are subclasses of one another should be considered compatible?

```py
from typing import Generic, TypeVar, TypedDict

T = TypeVar("T", bound=TypedDict)

class SerializesTo(Generic[T]):
    def serialize(self) -> T: ...

class SerializedAnimal(TypedDict):
    legs: int

class SerializedElephant(SerializedAnimal):
    trunkLength: float

class Animal(SerializesTo[SerializedAnimal]):
    pass

# Base classes of Elephant are mutually incompatible
# Base class "SerializesTo[SerializedElephant]" is incompatible with type "SerializesTo[SerializedAnimal]"
class Elephant(Animal, SerializesTo[SerializedElephant]):
    pass
```

I can't quite wrap my head around this error, as a SerializedElephant _is_ a SerializedAnimal.

**Comment by erictraut** [comment]
The problem is that `SerializesTo[SerializedElephant]` is not the same as `SerializesTo[SerializedAnimal]`, but `T` is invariant.

I can see a couple of ways to fix this:

1. Make `T` covariant. There's no reason why it cannot be in this case because it's used only in covariant locations.

```python
T = TypeVar("T", bound=TypedDict, covariant=True)
```

2. Make `Animal` generic.

```python
T2 = TypeVar("T2", bound=SerializedAnimal)

class Animal(SerializesTo[T2]):
    pass

class Elephant(Animal[SerializedElephant], SerializesTo[SerializedElephant]):
    pass
```

**Comment by Trimatix** [comment]
Fantastic, thank you so much :)
